### PR TITLE
(#1269726) extend inhibitor lock enforcement to root

### PR DESCRIPTION
**Pull request by jacekmigacz**
Some basic checks below:
```
# systemd-inhibit --what=shutdown --mode=block sleep infinity &
[1] 941
```
```
# systemctl --check-inhibitors=yes reboot
Operation inhibited by "sleep infinity" (PID 941 "systemd-inhibit", user root), reason is "Unknown reason".
User vagrant is logged in on sshd.
Please retry operation after closing inhibitors and logging out other users.
Alternatively, ignore inhibitors and users with 'systemctl reboot -i'.
```

With all these changes from `SD_BUS_METHOD` to `SD_BUS_METHOD_WITH_NAMES`; also `SD_BUS_SIGNAL` has become `SD_BUS_SIGNAL_WITH_NAMES`. It was not obligatory, but it is more consistent now.

### CODE REVIEW

**Comment by mrc0mmand** [comment]
Don't mind the GCC job fails, there's something wrong with the API atm, and it's trying to checkout a different PR (which is already merged). I'll check if there's something I can do.

**Comment by mrc0mmand** [comment]
@jacekmigacz could you, please, force-push this PR again? Just `git commit --am --no-ed && git push -f origin bz1269726-extend-inhibitor-lock-enforcement-to-root` should do the trick. For some reason GH actions have wrong metadata and keep trying to fetch the original PR (#193) instead of this one.

**Review by jamacku — COMMENTED** [review]
I think this PR should be open against the ``master`` branch. We usually open PR against specific branches only when we are dealing with Z-stream updates.

**Comment by mrc0mmand** [comment]
Yeah, I was just about to ask if we've already moved to the z-stream branches with the updates. The CI fails because the 8.5.0 branch lacks the necessary patches. I'll change the base of this PR to `master`, so @jacekmigacz doesn't have to open a new one.

File: src/login/logind-dbus.c [hunk 1]
```diff
@@ -1765,20 +1773,32 @@ static int method_do_shutdown_or_sleep(
...
+                r = sd_bus_message_read(message, "b", &interactive);
+
         if (r < 0)
                 return r;
 
+        if (with_flags && (flags & ~SD_LOGIND_SHUTDOWN_AND_SLEEP_FLAGS_PUBLIC))
+                return sd_bus_error_setf(error, SD_BUS_ERROR_INVALID_ARGS,
+                                         "Invalid flags parameter");
+
+        SET_FLAG(flags, SD_LOGIND_INTERACTIVE, interactive);
```
Comment (dtardon) [inline src/login/logind-dbus.c]: Hmm, this seems wrong to me. If `with_flags == true`, then `interactive == false` here, so the flag is being unconditionally reset... Or am I missing something?

Comment (dtardon) [reply src/login/logind-dbus.c]: I guess it wouldn't hurt to backport https://github.com/systemd/systemd/pull/18436 as well--that fixes the problem, if there indeed is one...

**Review by dtardon — APPROVED** [review]
LGTM

**Pull request merged**